pkg/core/resources/apis/hostnamegenerator/api/v1alpha1/zz_generated.deepcopy.go

```diff
@@ -121,9 +121,13 @@ func (in *LabelSelector) DeepCopyInto(out *LabelSelector) {
 	*out = *in
 	if in.MatchLabels != nil {
 		in, out := &in.MatchLabels, &out.MatchLabels
-		*out = make(map[string]string, len(*in))
-		for key, val := range *in {
-			(*out)[key] = val
+		*out = new(map[string]string)
+		if **in != nil {
+			in, out := *in, *out
+			*out = make(map[string]string, len(*in))
+			for key, val := range *in {
+				(*out)[key] = val
```

Comment:
This confuses me. Do you think you could simplify this double dereferences? It can be a followup if you want